pkg/k8shandler/migrations/kibana5to6.go

```diff
@@ -0,0 +1,439 @@
+	}
+
+	if curSett != nil {
+		if curSett.Index != nil {
+			if curSett.Index.Blocks.Write {
+				logrus.Infof("skipping setting index %q to read-only because already completed", kibanaIndex)
+				return nil
+			}
+		}
+	}
+
+	settings := &estypes.IndexSettings{
+		Index: &estypes.IndexingSettings{
+			Blocks: &estypes.IndexBlocksSettings{
+				Write: true,
+			},
+		},
+	}
+
+	if err := mr.esClient.UpdateIndexSettings(kibanaIndex, settings); err != nil {
+		return fmt.Errorf("failed to set index %q to read only: %s", kibanaIndex, err)
+	}
+	return nil
+}
+
+func (mr *migrationRequest) createNewKibana6Index() error {
+	curIndex, err := mr.esClient.GetIndex(kibana6Index)
+	if err != nil {
+		return fmt.Errorf("failed to get index for %q: %s", kibana6Index, err)
+	}
+
+	if curIndex != nil {
+		if curIndex.Name == kibana6Index {
+			logrus.Infof("skipping creating index %q anew because already existing", kibana6Index)
+			return nil
+		}
+	}
+
+	mappings := make(map[string]interface{})
+	err = json.Unmarshal([]byte(kibana6IndexMappings), &mappings)
+	if err != nil {
+		return fmt.Errorf("failed to read kibana 6 mappings: %s", err)
+	}
+
+	index := &estypes.Index{
+		Name: kibana6Index,
+		Settings: estypes.IndexSettings{
+			NumberOfShards: 1,
+			Index: &estypes.IndexingSettings{
+				Format: 6,
+				Mapper: &estypes.IndexMapperSettings{
+					Dynamic: false,
+				},
+				Mapping: &estypes.IndexMappingSettings{
+					SingleType: true,
+				},
+			},
+		},
+		Mappings: mappings,
+	}
+
+	if err := mr.esClient.CreateIndex(kibana6Index, index); err != nil {
+		return fmt.Errorf("failed to create new index %q: %s", kibana6Index, err)
+	}
+	return nil
+}
+
+func (mr *migrationRequest) reIndexIntoKibana6() error {
+	if mr.migrationCompleted() {
+		logrus.Infof("skipping re-indexing %q to %q because already completed", kibanaIndex, kibana6Index)
+		return nil
+	}
+
+	err := mr.esClient.ReIndex(kibanaIndex, kibana6Index, kibanReIndexScript, "painless")
+	if err != nil {
+		return fmt.Errorf("failed to reindex kibana6: %s", err)
+	}
+	return nil
+}
+
+func (mr *migrationRequest) aliasKibana() error {
+	if mr.migrationCompleted() {
+		logrus.Infof("skipping aliasing %q to %q because alias existing", kibanaIndex, kibana6Index)
+		return nil
+	}
+
+	actions := estypes.AliasActions{
+		Actions: []estypes.AliasAction{
+			{
+				Add: &estypes.AddAliasAction{
+					Index: kibana6Index,
+					Alias: kibanaIndex,
+				},
+			},
+			{
+				RemoveIndex: &estypes.RemoveAliasAction{
+					Index: kibanaIndex,
+				},
+			},
+		},
+	}
+
+	if err := mr.esClient.Aliases(actions); err != nil {
+		return fmt.Errorf("failed to change alias %s to %s: %s", kibanaIndex, kibana6Index, err)
+	}
+	return nil
+}
+
+const (
+	kibanReIndexScript = `
+ctx._source = [ ctx._type : ctx._source ]; ctx._source.type = ctx._type; ctx._id = ctx._type + ":" + ctx._id; ctx._type = "doc";
+`
+
+	kibana6IndexMappings = `
```

Comment:
is this something we need to inject or can we let Kibana populate this itself once its index is created?

Reply:
TBH, the mappings are grabbed from the [migration guide]https://www.elastic.co/guide/en/kibana/6.0/migrating-6.0-index.html) suggested by Elastic. I believe the need to be in the payload since kibana cannot create the index itself in the absence of xpack. WDYT?